### ISSUE TITLE
New filter option: Creation date as threshold/start date

### ISSUE THREAD

**Issue by wldmr** [mpcjanssen/simpletask-android#695]
I don’t really care about creation dates on my tasks, so I use them as start dates. ST already provides a filter option to treat creation dates as due dates. Can a similar option be added so that creation dates are treated as start dates?

**Comment by mpcjanssen** [comment]
There is a filter setting on the other tab to use create as threshold which should do this.

**Comment by smichel17** [comment]
>ST already provides a filter option to treat creation dates as due dates. 

AFAIK there is no way to treat create dates as due dates. You can treat them as threshold dates using the setting @mpcjanssen mentioned in the `OTHER` tab of the filter.

**Comment by wldmr** [comment]
> There is a filter setting on the other tab to use create as threshold which should do this.

I don’t see it. I do see an option to treat creation dates as due dates, but not as threshold dates (see below).

> AFAIK there is no way to treat create dates as due dates.

This is weird. I’m using the german translation. The last item on the `OTHER` tab says „Erstellungsdatum als Fälligkeit“ which means “Creation date as due date”. HOWEVER: Selecting it seems to do nothing (I guess it might affect sorting, but it certainly doesn’t color items according to creation date being used like a due date).

Maybe there is a misunderstanding here. I’d like to hide items with a future creation date. You say that this option exists, but I can’t find it. I’m using Simpletask Cloudless v8.3.1, which I *think* is the latest version. I’m confused. o_O

**Comment by smichel17** [comment]
v8.3.1 is the latest version. It sounds like this is an issue with the translation. The last item on the `OTHER` tab should be `Create is threshold`, which makes the app use creation date as threshold date, if there is no threshold set.

If you'd like to help change this, you can contribute a new translation over [here](https://hosted.weblate.org/engage/simpletask/).

**Comment by wldmr** [comment]
Done, thanks for directing me there. From the looks of it there seems to be quite some confusion between the terms threshold and due date in the translation. I corrected some of that as well.

That said, I want to reiterate that these options *still* don’t seem to work as I would expect: If I *deselect* `Show tasks with threshold date in future` and *select* `Create is threshold`, I expect all items with a future creation date to be filtered out. But that does not happen. Am I misunderstanding that feature, or is the relation between these options buggy?

**Comment by smichel17** [comment]
"On unexpected input, a program may behave in unexpected ways."

How (or maybe the better question is, why) would you end up with a future creation date?

**Comment by wldmr** [comment]
I end up with a future creation date the same way I end up with a future threshold date: by choice.

Like I said, I use the creation date as a start date, or more specifically as a next-action marker. Because to me it doesn’t matter when I wrote something down, what matters is when I can start doing it (no date / date in the future → task not relevant yet). When I’m ready to start on a task, I add a date. Then, when I check it off, the end date gets added and I can see at a glance how long I procrastinated on a task.

My main reason for doing it that way is that in Vim (using the todo.txt plugin), setting the creation date to today is a two keystroke action (<leader>d) whereas adding a threshold is an 8 keystroke action (t:date<tab>).

I see why you would be reluctant to add that feature though; it is kind of a bending of the todo.txt rules and might be confusing for users. I suppose what I want can be achieved via Lua, so I’ll stop bugging you.

That said, from the wording of the option (`Create is threshold`), I hope you can see why one might expect the behavior I described (“treat the creation date as a threshold date”). So may I suggest rewording that option a little (like adding “for sorting purposes” or somesuch)? Or, you know, *don’t* reword it and change the behavior? ;)

Alright, I’ll stop bugging you. Keep up the great work!

**Comment by smichel17** [comment]
You're not bugging me. Although it is difficult to reason about what the expected behavior would be with a creation date in the future, I do agree that it would make sense to have `create is threshold` hide tasks with a threshold date in the future.

There needs to be some backend improvement to make development in area easier, which starts with #617. Also, when the semester is over. 

There are a lot of things about filtering that I don't love, but it does work right now overall so I'm hesitant to start changes right now.

**Comment by Vijayanth-Reddy-K** [comment]
IMHO, the behaviour of the `Create is threshold` option should be made explicit. Otherwise, the existing option should be renamed `Create is threshold (implicit)` and a further `Create is threshold (explicit)` be added.

By explicit behaviour, I mean, if the option is selected, the app cycles through all the existing tasks, and if the threshold date is missing, it is explicitly added as t:<creation_date> by the app. And if a new task is created but does not have a threshold date, it is added before being saved.

The reason for this: If I have a recurring task (say rec: +1d) which also has a due date, completing the task increments the due date, but the threshold date cannot be changed because it is the same as the creation date. And if the `Show tasks with threshold date in future` option is unchecked, the task is still visible, although that might be because of the behaviour outlined above.

**Comment by smichel17** [comment]
Imo, creation date should be incremented along with threshold and due date, so this is a separate issue, with recurring tasks, not one with  `Create is threshold`.

I do think create is threshold should be in the sort tab, though. Just need to find a place for it.

**Comment by mpcjanssen** [comment]
Both functionalities requested:

- Sorting on a specific date with other dates as fallback
- Adding explicit dates if they are not specified.

can be implemented using the `onAdd` and `onSort` callbacks. As a result, I am not going to add any other options in the "Other" tab.
